src/main/java/com/training/tinyurl/service/TinyUrlServiceImpl.java

```diff
@@ -1,71 +1,40 @@
-import com.training.tinyurl.dto.RegistrationReqDto;
 import com.training.tinyurl.dto.UpdateUrlDto;
 import com.training.tinyurl.entity.TinyUrlEntity;
 import com.training.tinyurl.entity.UserInfoEntity;
 import com.training.tinyurl.exceptionhandler.MongoApiException;
+import com.training.tinyurl.exceptionhandler.QuotaLimitExceededException;
+import com.training.tinyurl.exceptionhandler.TooManyCollisionsException;
 import com.training.tinyurl.repo.TinyUrlRepo;
 import com.training.tinyurl.repo.UserInfoRepo;
 import com.training.tinyurl.security.AppUserDetails;
 import com.training.tinyurl.util.TinyUrlGenerator;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
-import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import java.security.NoSuchAlgorithmException;
 import java.util.Optional;
 
 @Service
 @Slf4j
 public class TinyUrlServiceImpl implements ITinyUrlService{
-
     private final UserInfoRepo userInfoRepo;
-    private final PasswordEncoder encoder;
     private final TinyUrlRepo tinyUrlRepo;
-
-    public TinyUrlServiceImpl(UserInfoRepo userInfoDb, PasswordEncoder encoder, TinyUrlRepo tinyUrlRepo) {
+    public TinyUrlServiceImpl(UserInfoRepo userInfoDb, TinyUrlRepo tinyUrlRepo) {
         this.userInfoRepo = userInfoDb;
-        this.encoder = encoder;
         this.tinyUrlRepo = tinyUrlRepo;
     }
 
-    @Override
-    public void createNewUser(RegistrationReqDto request) throws MongoApiException {
-        request.setEmail(request.getEmail().toUpperCase());
-        if(!userInfoRepo.existsById(request.getEmail())){
-            userInfoRepo.save(new UserInfoEntity(
-                    request.getEmail(),
-                    encoder.encode(request.getPassword()),
-                    request.getAccountType()));
-                    return;
-        }
-        log.error("User with same email already exists");
-        throw new MongoApiException(HttpStatus.CONFLICT,"User with same email already exists");
-    }
-
-    @Override
-    public void upgradePlan(AppUserDetails userDetails){
-        userInfoRepo.save(new UserInfoEntity(
-                userDetails.getEmail(),
-                userDetails.getPassword(),
-                AccountType.PREMIUM
-        ));
-        // we need to login again once the authorities are changed
-        logoutUser();
-    }
-
-    @Override
-    public void logoutUser() {
-        SecurityContextHolder.getContext().setAuthentication(null);
-    }
-
-    private boolean isAllowedToGenUrl(AppUserDetails user){
+    private boolean isAllowedToGenerateUrl(AppUserDetails user){
         Optional<UserInfoEntity> userInfo = userInfoRepo.findById(user.getEmail());
-        return userInfo.get().getAccountType().toString().equals("PREMIUM")
-                || userInfo.get().getUsedQuota() < 100;
+        final int MAX_QUOTA = 100;
```

Comment:
should be moved to a constant file or much better to application.properties file.
final int doesn't make it constant. Constants are global variables and shared (static).
Local variable is separate for each thread (here user request).